src/app.js

```diff
@@ -1,342 +1,231 @@
+  _isDestroyed: false,
 
   /**
-   * Stops `childApps` if allowed by child
+   * Set to true if a parent `App` should not be able to destroy this `App`.
    *
-   * @private
-   * @method _stopChildApps
-   * @memberOf App
+   * @type {Boolean|Function}
+   * @default false
    */
-  _stopChildApps: function() {
-    _.each(this._childApps, function(childApp) {
-      if(_.result(childApp, 'stopWithParent')) {
-        childApp.stop();
-      }
-    });
-  },
+  preventDestroy: false,
 
   /**
-   * Destroys `childApps` if allowed by child
+   * Set to true if `App` should be started after it is initialized.
    *
-   * @private
-   * @method _destroyChildApps
-   * @memberOf App
+   * @type {Boolean|Function}
+   * @default false
    */
-  _destroyChildApps: function() {
-    _.each(this._childApps, function(childApp) {
-      if(!_.result(childApp, 'preventDestroy')) {
-        childApp.destroy();
-      }
-    });
-  },
+  startAfterInitialized: false,
 
   /**
-   * Internal helper to instantiate and `App` from on `Object`
+   * Set to true if `App` should be started after its parent starts.
    *
-   * @private
-   * @method _buildAppFromObject
-   * @memberOf App
-   * @param {Object} appConfig - `AppClass` and any other option for the `App`
-   * @returns {App}
+   * @type {Boolean|Function}
+   * @default false
    */
-  _buildAppFromObject: function(appConfig) {
-    const AppClass = appConfig.AppClass;
-    const options = _.omit(appConfig, 'AppClass');
-
-    return this.buildApp(AppClass, options);
-  },
+  startWithParent: false,
 
   /**
-   * Helper for building an App and return it
+   * Set to false if `App` should not stop after its parent stops.
    *
-   * @private
-   * @method _buildApp
-   * @memberOf App
-   * @param {App} AppClass - An App Class
-   * @param {Object} AppClass - Optionally passed as an appConfig Object
-   * @param {Object} [options] - options for the AppClass
-   * @returns {App}
+   * @type {Boolean|Function}
+   * @default true
    */
-  _buildApp: function(AppClass, options) {
-    if(_.isFunction(AppClass)) {
-      return this.buildApp(AppClass, options);
-    }
-    if(_.isObject(AppClass)) {
-      return this._buildAppFromObject(AppClass);
-    }
-  },
+  stopWithParent: true,
 
   /**
-   * Build an App and return it
-   * Override for dynamic App building
-   *
    * @public
-   * @method buildApp
-   * @memberOf App
-   * @param {App} [AppClass] - An App Class
-   * @param {Object} [options] - options for the AppClass
-   * @returns {App}
+   * @constructs App
+   * @param {Object} [options] - Settings for the App.
+   * @param {Boolean} [options.startWithParent]
+   * @param {Boolean} [options.stopWithParent]
+   * @param {Boolean} [options.startAfterInitialized]
+   * @param {Boolean} [options.preventDestroy]
    */
-  buildApp: function(AppClass, options) {
-    return new AppClass(options);
+  constructor(options = {}) {
+    _.bindAll(this, 'start', 'stop');
+
+    const pickOptions = [
```

Comment:
hmm... maybe not for now, but I wonder if this'd be nicer if we just defined this as.. maybe `classOptions` or something just above the definition.  Almost makes the file more documenting.